### ISSUE TITLE
Broken auto-import of utils functions when used in object shorthand notation

### ISSUE THREAD

**Issue by lehni** [nuxt/nuxt#21807]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.15.0`
- Nuxt Version:     `3.6.1`
- Nitro Version:    `2.5.1`
- Package Manager:  `yarn@3.6.0`
- Builder:          `vite`
- User Config:      `workspaceDir`, `telemetry`, `server`, `runtimeConfig`, `app`, `modules`, `pinia`, `router`, `routeRules`, `css`, `postcss`, `vite`, `nitro`, `imports`, `components`, `ignore`, `hooks`, `watch`
- Runtime Modules:  `@pinia/nuxt@0.4.11`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/object-shorthand-auto-import?file=utils%2FautoImportedMethod.js

### Describe the bug

Up to v3.5.3, it was possible to correctly auto-import utils functions by using their name in object shorthand notation

Nuxt v3.6.1 doesn't seem to correctly auto-import such functions anymore, as illustrated by the reproduction.

### Logs

```shell-script
autoImportedMethod is not defined

at eval (./app.vue:11:5)
at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:351:5)
at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:207:14)
at async ViteNodeRunner.dependencyRequest (./node_modules/vite-node/dist/client.mjs:241:12)
at async eval (./virtual:nuxt:/home/projects/object-shorthand-auto-import/.nuxt/app-component.mjs:1:216)
at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:351:5)
at async ViteNodeRunner.cachedRequest (./node_modules/vite-node/dist/client.mjs:207:14)
at async ViteNodeRunner.dependencyRequest (./node_modules/vite-node/dist/client.mjs:241:12)
at async eval (./node_modules/nuxt/dist/app/components/nuxt-root.vue:5:31)
at async ViteNodeRunner.directRequest (./node_modules/vite-node/dist/client.mjs:351:5)
```

**Comment by danielroe** [comment]
cc: @antfu

**Comment by antfu** [comment]
It was a regression, investigating

**Comment by antfu** [comment]
Fixed in `unimport@3.0.11`, upgrade your lock file should fix this. Thanks.

**Comment by lehni** [comment]
thanks @antfu !
